lib/event-decoder.js

```diff
@@ -0,0 +1,49 @@
+const avro = require('avsc')
+
+const logger = require('./logger')
+const platformApi = require('./platform-api/requests')
+
+const CACHE = {}
+
+async function _getSchema (schemaName) {
+  // schema in cache; just return it as a instant promise
+  if (CACHE[schemaName]) {
+    logger.debug(`Already have ${schemaName} schema`)
+    return Promise.resolve(CACHE[schemaName])
+  }
+
+  // Fetch schema and parse it as an AVSC decoder
+  const { data: schemaObject } = await platformApi.getSchema(schemaName)
```

Comment:
Doesn't this assign the value of the `data` property to a new variable called `schemaObject`? (That is, your `schemaObject` takes on the complete value of response.data) I think this should actually be `const { data: { schemaObject: schemaObject } } = await ...` or `const { data: { schemaObject } } = await ...` for brevity

Reply:
But also, looks like `lib/platform-api/requests#getSchema` is returning `response.data`, so no need to destructure the `data` prop

Reply:
yes! good catch on both counts